### PR TITLE
Delete VPC and subnets

### PR DESCRIPTION
**Pull request by nprokopic**
This PR:

- adds/changes/removes etc

### Testing

Description on how aws-vpc-operator can be tested.

- [ ] fresh install works
- [ ] upgrade from previous version works
- [ ] aws-vpc-operator manages only VPCs that are not managed by CAPA

#### Other testing

Description of features to additionally test for aws-vpc-operator installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] AWS VPC works after aws-vpc-operator upgrade

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: pkg/aws/subnets/reconciler_delete.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/aws-vpc-operator/pkg/aws"
+	"github.com/giantswarm/aws-vpc-operator/pkg/errors"
+)
+
+func (r *reconciler) ReconcileDelete(ctx context.Context, request aws.ReconcileRequest[[]aws.DeletedCloudResourceSpec]) (err error) {
```
Comment (AverageMarcus) [inline pkg/aws/subnets/reconciler_delete.go:13]: Why do the delete reconcilers use these generics types for the request? But the normal doesn't? Why is it a slice?

Comment (nprokopic) [reply pkg/aws/subnets/reconciler_delete.go:13]: > Why do the delete reconcilers use these generics types for the request? But the normal doesn't? 

Gradual refactoring towards all reconcilers having the same approach with generics (so there is less copy paste code).

Comment (nprokopic) [reply pkg/aws/subnets/reconciler_delete.go:13]: > Why is it a slice?

Specifying the subnets that should be deleted.

Comment (nprokopic) [reply pkg/aws/subnets/reconciler_delete.go:13]: It will all make sense in a PR or two :smile:

**Pull request merged**